include/deal.II/lac/sparse_direct.h

```diff
@@ -304,9 +304,12 @@ class SparseDirectUMFPACK : public Subscriptor
         "want to check your assembly procedure. Similarly, a "
         "matrix can be rank deficient if you forgot to apply the "
         "appropriate boundary conditions. For example, the "
-        "Laplace equation without boundary conditions has a "
-        "single zero eigenvalue and its rank is therefore "
-        "deficient by one."
+        "Laplace equation for a problem where only Neumann boundary "
+        "conditions are posed (or where you forget to apply Dirichlet "
+        "boundary conditions) has a one zero eigenvalue and its rank is "
```

Comment:
```suggestion
        "boundary conditions) has exactly one eigenvalue equal to zero and its rank is "
```
"has a one zero" doesn't look right and I think its clearer to not write "one zero".